TLM/TLM/Manager/Impl/OptionsManager.cs

```diff
@@ -193,125 +116,102 @@ public class OptionsManager
+            save[19] = (byte)(Options.timedLightsEnabled ? 1 : 0);
+            save[20] = (byte)(Options.customSpeedLimitsEnabled ? 1 : 0);
+            save[21] = (byte)(Options.vehicleRestrictionsEnabled ? 1 : 0);
+            save[22] = (byte)(Options.laneConnectorEnabled ? 1 : 0);
+            save[23] = (byte)(Options.junctionRestrictionsOverlay ? 1 : 0);
+            save[24] = (byte)(Options.junctionRestrictionsEnabled ? 1 : 0);
+            save[25] = (byte)(Options.parkingAI ? 1 : 0);
+            save[26] = (byte)(Options.preferOuterLane ? 1 : 0);
+            save[27] = (byte)(Options.individualDrivingStyle ? 1 : 0);
+            save[28] = (byte)(Options.evacBussesMayIgnoreRules ? 1 : 0);
+            save[29] = (byte)(Options.instantEffects ? 1 : 0);
+            save[30] = (byte)(Options.parkingRestrictionsEnabled ? 1 : 0);
+            save[31] = (byte)(Options.parkingRestrictionsOverlay ? 1 : 0);
+            save[32] = (byte)(Options.banRegularTrafficOnBusLanes ? 1 : 0);
+            save[33] = (byte)(Options.showPathFindStats ? 1 : 0);
+            save[34] = Options.altLaneSelectionRatio;
+            save[35] = (byte)Options.vehicleRestrictionsAggression;
+            save[36] = (byte)(Options.trafficLightPriorityRules ? 1 : 0);
+            save[37] = (byte)(Options.realisticPublicTransport ? 1 : 0);
+            save[38] = (byte)(Options.turnOnRedEnabled ? 1 : 0);
+            save[39] = (byte)(Options.allowNearTurnOnRed ? 1 : 0);
+            save[40] = (byte)(Options.allowFarTurnOnRed ? 1 : 0);
+            save[41] = (byte)(Options.automaticallyAddTrafficLightsIfApplicable ? 1 : 0);
+
+            save[42] = OptionsMassEditTab.RoundAboutQuickFix_StayInLaneMainR.Save();
+            save[43] = OptionsMassEditTab.RoundAboutQuickFix_StayInLaneNearRabout.Save();
+            save[44] = OptionsMassEditTab.RoundAboutQuickFix_DedicatedExitLanes.Save();
+            save[45] = OptionsMassEditTab.RoundAboutQuickFix_NoCrossMainR.Save();
+            save[46] = OptionsMassEditTab.RoundAboutQuickFix_NoCrossYieldR.Save();
+            save[47] = OptionsMassEditTab.RoundAboutQuickFix_PrioritySigns.Save();
+
+            save[48] = OptionsMassEditTab.PriorityRoad_CrossMainR.Save();
+            save[49] = OptionsMassEditTab.PriorityRoad_AllowLeftTurns.Save();
+            save[50] = OptionsMassEditTab.PriorityRoad_EnterBlockedYeild.Save();
+            save[51] = OptionsMassEditTab.PriorityRoad_StopAtEntry.Save();
+
+            save[52] = OptionsMassEditTab.RoundAboutQuickFix_KeepClearYieldR.Save();
+            save[53] = OptionsMassEditTab.RoundAboutQuickFix_RealisticSpeedLimits.Save();
+            save[54] = OptionsMassEditTab.RoundAboutQuickFix_ParkingBanMainR.Save();
+            save[55] = OptionsMassEditTab.RoundAboutQuickFix_ParkingBanYieldR.Save();
+
+            save[56] = OptionsVehicleRestrictionsTab.NoDoubleCrossings.Save();
+            save[57] = OptionsVehicleRestrictionsTab.DedicatedTurningLanes.Save();
```

Comment:
@aubergine10 How could I missed that! This will fail. You are setting array to size(capacity) 57 and here you are trying to set **57th index (58th element)**. Index out of range guaranteed.

Reply:
Yup, picked that up in another PR just now - will send a commit to the What's New PR #1332

Reply:
Done: https://github.com/CitiesSkylinesMods/TMPE/pull/1339/commits/d784e164c54aa4ea699d42ff0dd184efff952a3c